### PR TITLE
Refactor py_esdl_from_file to use Path and str

### PR DESCRIPTION
**Pull request by AshrafAlmohagryDeltares**
I have added a path to the file_path entry to accept both string or Path. Please review it.

### CODE REVIEW

File: src/omotes_simulator_core/infrastructure/utils.py [hunk 1]
```diff
@@ -19,17 +19,21 @@
...
 
 
-def pyesdl_from_file(file_path: str) -> EnergySystemHandler:
+def pyesdl_from_file(file_path: str | Path) -> EnergySystemHandler:
     """
     Loads esdl file into memory and returns a handle to be able to use this file.
 
     Please note that it is not checked if the file is a valid esdl file.
-    :param file_path: string pointing to the esdl file to be loaded into memory
+    :param file_path: string or Path pointing to the esdl file to be loaded into memory
```
Comment (vanmeerkerk) [inline src/omotes_simulator_core/infrastructure/utils.py:27]: Use the following format: 

:param str | Path file_path: Location of the ESDL file to be loaded into memory.

**Review by vanmeerkerk — CHANGES_REQUESTED** [review]
See comment.

**Comment by AshrafAlmohagryDeltares** [comment]
I have fixed the format. Please review it.

**Review by vanmeerkerk — APPROVED** [review]
Checked

**Pull request merged**